README.md

```diff
@@ -1,2 +1,422 @@
+```bash
+# run this command inside the flutter project directory `demo_app/`
+flutter pub add camera intl;
+```
+4. In order to use the camera, we need to set the `minSdkVersion` in `android/app/build.gradle`:
+```groovy
+// inside the 'android/app/build.gradle' set the 'minSdkVersion' to 21
+    defaultConfig {
+        ...
+        minSdkVersion 21
+        ...
+    }
+```
+5. (optional) The app is ready to be built, but it will throw deprecation warnings. If you want to clear the deprecation warnings, follow this step. [See this issue](https://github.com/flutter/flutter/issues/89578#issuecomment-945916643)
+```yaml
+# change the respective part inside the 'pubspec.yaml'
+dependencies:
+  camera:
+    git:
+      url: https://github.com/flutter/plugins
+      path: packages/camera/camera
+      ref: 9e46048ad2e1f085c1e8f6c77391fa52025e681f
+```
+
+6. [Run the Android Emulator](https://developer.android.com/studio/run/emulator)
+
+7. Run the demo app:
+ ```bash
+flutter run
+ ```
+
+8. **Set the device to landscape mode**. askui currently works only with landscape mode, meaning that the width of the captured image should be bigger than the height. If you test in portrait mode, then it will run into an issue.
+
+Now you should see the demo app running on your android device.
+
+
+### 2. Setup ADBKeyboard
+In this example, we are going to automate the typing in the android device. In order to let askui fluently type as desired, we will use a virtual keyboard that handles the keyboard input via adb: [ADBKeyboard.apk](https://github.com/senzhk/ADBKeyBoard)
+
+1. Download the ADB-Keyboard package *(Important: Version 2.0)*: <br>[Link to Github Repo](https://github.com/senzhk/ADBKeyBoard/releases/tag/v2.0)
+2. Unzip it.
+3. Find your device:
+```bash
+# make sure that your android device is connected, and the USB debugging mode is enabled.
+adb devices
+```
+4. Install the ADBkeyboard on the device:
+```bash
+# inside ADBKeyBoard-2.0/
+adb -s <your device id> install ADBKeyboard.apk
+```
+5. Configure the ADB Keyboard:
+```bash
+# inside ADBKeyBoard-2.0/
+adb -s <your device id> shell settings put secure default_input_method com.android.adbkeyboard/.AdbIME
+```
+6. Enable the ADB Keyboard:
+```bash
+# inside ADBKeyBoard-2.0/
+adb -s <your device id> shell ime enable com.android.adbkeyboard/.AdbIME
+```
+7. To check if it is enabled: <br>Click on a textfield in an app and see if the `ADB Keyboard {ON}` notification is shown at the bottom of the screen.
+
+
+### 3. Setup askui
+1. Setup askui by following [these steps](https://docs.askui.com/docs/general/Getting%20Started/getting-started).
+2. We need to run the uiController directly with an extra argument to specify the runtime mode, as the current version of askui(ver. 0.3.2) doesn't provide the api for running it with the runtime argument yet. <br><br>From within your npm project path, go to the directory that contains the askui-ui-controller binary, and run `./askui-ui-controller -r android`
```

Comment:
More nitpicking:

uiController -> UiController (Derived from the class in TypeScript)